unittests/DOSGuard.cpp

```diff
@@ -81,14 +93,75 @@ class DOSGuardTest : public NoLoggerFixture
 {
 protected:
     Config cfg{json::parse(JSONData)};
+    Config cfg_static_list{json::parse(JSONData3)};
     FakeSweepHandler sweepHandler;
+    FakeSweepHandler sweepHandler_static_list_ipv4;
     BasicDOSGuard<FakeSweepHandler> guard{cfg, sweepHandler};
+    BasicDOSGuard<FakeSweepHandler> guard_mixed{
+        cfg_static_list,
+        sweepHandler_static_list_ipv4};
 };
 
 TEST_F(DOSGuardTest, Whitelisting)
 {
+    // Test cases for individual IPv4 address
     EXPECT_TRUE(guard.isWhiteListed("127.0.0.1"));
     EXPECT_FALSE(guard.isWhiteListed(IP));
+
+    // Test cases for list of IPv4 addresses
+    EXPECT_TRUE(guard_mixed.isWhiteListed("127.0.0.1/24"));
```

Comment:
Looks like you did not fully understand the purpose of the whitelist and what it means to check whether an IP is within a subnet or not.

Here is a test that should be passing and is currently failing:

```cpp
// assume "whitelist": ["192.168.0.0/24"]
EXPECT_TRUE(guard_mixed.isWhiteListed("192.168.0.1"));
EXPECT_FALSE(guard_mixed.isWhiteListed("192.168.1.1"));
```

I believe I told you before that `isWhiteListed` can never get a mask, only a real IP, hence this test makes no sense.
No need to review the rest of the code for now because it fundamentally does not do what it should.

Reply:
@godexsoft I went off an incorrect assumption regarding intended scope of isWhitelisted. Functionality for checking subnets in different methods and assumed those to be public, but will integrate that functionality + testing live into the isWhitelisted (now clarified as the **only** public method to users) and getWhitelist methods in the next iteration. Standby